sequence_processing_pipeline/Pipeline.py

```diff
@@ -459,21 +459,50 @@ def get_sample_names(self, project_name=None):
 
     def _get_sample_names_from_sample_sheet(self, project_name):
         if project_name is None:
-            return [x.Sample_Name for x in self.sample_sheet.samples]
+            if 'orig_name' in self.sample_sheet.samples[0]:
+                # return set as a list to match non-replicate results.
+                return sorted(list({x.orig_name for x in
+                                    self.sample_sheet.samples}))
+            else:
+                return [x.Sample_Name for x in self.sample_sheet.samples]
         else:
             # Since the project-name is stored in an internal variable
             # in a third-party library, convert the data structure to
             # JSON using the exposed method and obtain from the result.
             jsn = json_loads(self.sample_sheet.to_json())
```

Comment:
This comment seems very surprising. Is it _really_ necessary to serialize / deserialize to access the project name?

Reply:
Agreed, especially when considering the recent changes, a better choice seems obvious.